wsd/ClientSession.cpp

```diff
@@ -2049,6 +2091,64 @@ bool ClientSession::handleKitToClientMessage(const std::shared_ptr<Message>& pay
                                               payload->data().size() - firstLine.size() - 1);
             return forwardToClient(payload);
         }
+        else if (tokens.equals(0, "extractedlinktargets:"))
+        {
+            LOG_TRC("Sending extracted link targets response.");
+
+            const std::string stringJSON = payload->jsonString();
+
+            http::Response httpResponse(http::StatusCode::OK);
+            httpResponse.set("Last-Modified", Util::getHttpTimeNow());
+            httpResponse.set("X-Content-Type-Options", "nosniff");
+            httpResponse.setBody(stringJSON, "application/json");
+            _saveAsSocket->sendAndShutdown(httpResponse);
+
+            LOG_TRC("Removing export-link-targets ClientSession.");
+
+            // Remove us.
+            docBroker->removeSession(client_from_this());
+
+            // Now terminate.
+            docBroker->stop("Finished extracting link targets handler.");
```

Comment:
Neither of these methods is the proper way to do this. If we want to end the docBroker instance, we should handle the message there and properly call `closeDocument()`.